Chirp/Networking/NetworkService.swift

```diff
@@ -77,5 +84,57 @@ struct NetworkService {
         .eraseToAnyPublisher()
     }
     
+    func createUser(request: CreateUserRequest) -> AnyPublisher<User, Error> {
+        Deferred {
+            Future { promise in
+                db.collection(Table.users.rawValue).document(request.userID).setData([
+                    Key.name.rawValue: request.name,
+                    Key.phoneNumber.rawValue: request.phoneNumber,
+                    Key.profilePicture.rawValue: request.profilePicture ?? ""
```

Comment:
Even though this is good and works. We also have an extension of Encodable that you can use by saying `request.asDictionary` This will encode the entire struct into the type [String: Any].

You don't need to change yours. It's just a side note